app/controllers/checkout_controller.rb

```diff
@@ -54,6 +54,9 @@ def update
       else
         update_failed
       end
+    elsif @order.errors.messages[:items_removed].present?
```

Comment:
this action is going too wild. We'll have to do something about it soon.

Reply:
>  simply passing a regular error onto the update_failed method at the end made the checkout page freeze up. :/

Did you find out why? I'm afraid there's too much going on in this action that we don't know. Anything we can do to clarify it, no matter how small it is, it will be great.